### PR TITLE
22 sensor daemon

### LINKED ISSUE #22: Implement daemon sensor data fetching
r/titlegore

~~Implement a ZMQ endpoint for the daemon to get sensor data from. Probably a SUB socket.~~

Implement a daemon to get sensor data from Si7021.

### PR DESCRIPTION
**Pull request by Zippoman924**
Implements a daemon that pulls temperature and humidity data from the Si7021 sensor. 

Also implements the Arduino code that the runs on the Arduino attached to the Raspberry Pi and the Si7021. 

**Important Note**
The Arduino code must be flashed to the Arduino using the Arduino app. When flashing the code you must follow these steps to properly include the Si7021 library from Adafruit. 
1. Open the .ino file with the Arduino editor
2. Select "Sketch" -> "Include Library" -> Manage Libraries...
3. Type "Si7021" into the search bar
4. Select the "Install" option on the "Adafruit Si7021 Library" result
5. Flash the Arduino with the temp sensor code

*closes #22*

### CODE REVIEW

**Review by Eli-Labell — APPROVED** [review]
It looks good to me.

File: tempsensor/makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+CC = g++
+LD = g++
+CCFLAGS = -Wall
+LDFLAGS = -lstdc++ -pthread -L ./target/debug/
+EXE = tempsensor
```
Comment (minifigone) [inline tempsensor/makefile:5]: call it `tempsensord`, that's daemon naming convention

File: tempsensor/sensor_daemon.hpp [hunk 2]
```diff
@@ -0,0 +1 @@
+int *nop(void);
```
Comment (minifigone) [inline tempsensor/sensor_daemon.hpp:1]: if you're not using `nop` for anything you can drop it, i had it in the other daemon as a placeholder until the thread functions got implemented.

**Review by minifigone — APPROVED** [review]
looks good. i'm guessing the json formatting will come with the networking tickets. my comments are pretty minor changes.

File: tempsensor/sensor_daemon.hpp (hunk 2 above)
Comment (Zippoman924) [reply tempsensor/sensor_daemon.hpp:1]: I just ended up removing the whole header file in the next commit to fix this.

File: tempsensor/makefile (hunk 1 above)
Comment (Zippoman924) [reply tempsensor/makefile:5]: Fixed it in the most recent commit.

**Pull request merged**